### DISCUSSION TITLE
504 error | serverless function | PRO member | maxDuration 15s Timeout | v13.5.6

### DISCUSSION THREAD

**Discussion by VeeShostak** [vercel/community#4567] [Help] [answered]
### Summary

Pro Member
I tried everything, same issue on 13.5.7-canary.13 and 13.5.6

vercel.json is in root, 
my function is in src/pages/api/exract-wl.tsx

I went bananas on the config to add every possible combination
```
{
  "functions": {
    "src/pages/api/*.js": {
      "maxDuration": 180
    },
    "src/pages/api/*.ts": {
      "maxDuration": 180
    },
    "pages/api/exract-wl.js": {
      "maxDuration": 180
    },
    "pages/api/exract-wl.ts": {
      "maxDuration": 180
    },
    "src/pages/api/exract-wl.js": {
      "maxDuration": 180
    },
    "src/pages/api/exract-wl.ts": {
      "maxDuration": 180
    }
  }
}

```

my api route ALSO has the max duration
```
export const dynamic = 'force-dynamic';
export const maxDuration = 180; // 3 min
export const runtime = 'nodejs'

export default async function POST(
    req: NextApiRequest,
    res: NextApiResponse
) {
```

I tried renaming the function and redeploying.

It always says 
`Maximum Execution Duration 15 seconds` in Vercel, and it dies at 15.01 seconds.

Currently in deep prayer 

### Steps to Reproduce

add your api route:
src/pages/api/exract-wl.tsx

Give it a max duration
```
export const dynamic = 'force-dynamic';
export const maxDuration = 180; // 3 min
export const runtime = 'nodejs'

export default async function POST(
    req: NextApiRequest,
    res: NextApiResponse
) {
```

also add a vercel.json config file to your root
```
{
  "functions": {
    "src/pages/api/*.js": {
      "maxDuration": 180
    },
    "src/pages/api/*.ts": {
      "maxDuration": 180
    },
    "pages/api/exract-wl.js": {
      "maxDuration": 180
    },
    "pages/api/exract-wl.ts": {
      "maxDuration": 180
    },
    "src/pages/api/exract-wl.js": {
      "maxDuration": 180
    },
    "src/pages/api/exract-wl.ts": {
      "maxDuration": 180
    }
  }
}

```

**Comment by VeeShostak** [comment, marked as answer]
Resolved with:
```
export const config = {
    maxDuration: 300, // 5 minutes.
};
```

**Reply by architectonic** [reply]
Running into the same issue, tried adding the export config lines with no result. Could you please clarify how you managed to increase the 15s limit?

**Reply by VeeShostak** [reply]
If you are on NextJS 13.2+
but your project is still using the old api routes,
```
i.e 

pages/api/test.tsx (Old)
VS
/api/test/route.ts (New)
```


You would have to export the maxDuration from a config object like this:

```
export const config = {
    maxDuration: 300, // 5 minutes.
};
export default async function extractWl(
    req: NextApiRequest,
    res: NextApiResponse
) {
...
```

This worked for me on 13.5.7-canary.13


That was the issue for one of my playground projects, I have since then switched to the new api routing and just exporting the maxDuration as in the Vercel docs now works as well

**Comment by Krak86** [comment]
It works on 13.5.6 with maxDuration: 300
Thanks!
